software/controller/lib/hal/spi_chain.tpp

```diff
@@ -222,14 +207,42 @@ void DaisyChain<MaxSlaves>::TransmitNextCommand() {
   }
 
   if(data_to_send){
-    SetupReception(receive_buffer_, num_slaves_);
+    spi_->SetupReception(receive_buffer_, num_slaves_);
+    // In case cs has not been high for long enough, delay sending the command
+    Time now = SystemTimer::singleton().now();
```

Comment:
This looks like a tasty little morsel to abstract, e.g.:
`SystemTimer::ensure_delay(Duration delay, Time since);`

Could also be easily mocked/faked for other tests.